### ISSUE TITLE
pcsclite ships a systemd service unit with an unreplaced ${prefix} variable

### ISSUE THREAD

**Issue by kirelagin** [NixOS/nixpkgs#148538]
### Describe the bug

The `pcsclite` package installs `pcscd.service` that contains an unconfigured `prefix` variable:

```text
$ cat /etc/systemd/system/pcscd.service | grep EnvironmentFile
EnvironmentFile=-${prefix}/etc/default/pcscd
```

This results in the following message in system journal:

```text
systemd[1]: /etc/systemd/system/pcscd.service:9: EnvironmentFile= path is not absolute, ignoring: ${prefix}/etc/default/pcscd
```

Obviously, not critical, but still this doesn’t sound right.

The [upstream file](https://salsa.debian.org/rousseau/PCSC/-/blob/e1b1590c7f32e0353404967f457643d122367772/etc/pcscd.service.in#L9) contains `EnvironmentFile=-@sysconfdir@/default/pcscd`, so it looks like `@sysconfdir@` gets replaced by literal `${prefix}/etc/`, but I am not sure where this happens or why.

### Steps To Reproduce
1. `nix build nixpkgs#pcsclite`
2. `grep EnvironmentFile result-bin/lib/systemd/system/pcscd.service`

### Expected behavior
There should be no unsubstituted variables in the installed unit.

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: [ pcsclite ]
# a list of nixos modules affected by the problem
module:
```

**Comment by kirelagin** [comment]
Looks like this line was added upstream in 1.9.2 and the version in nixpkgs was upgraded from 1.9.1 to 1.9.4 in October (#142080).

**Comment by kirelagin** [comment]
Ok, I am not particularly good with Autotools, but it seems to me that it is an upstream bug. The generate `./configure` does this replacement, but I think it is supposed to happen only in a context in which this `${prefix}` will be further expanded, not in plaintext files intended to be simply copied to their final install location.

**Comment by kirelagin** [comment]
Yes, it’s definitely an upstream issue. This usage of `@sysconfdir@` is explicitly called out as incorrect in Autconf docs: https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.70/html_node/Installation-Directory-Variables.html#Installation-Directory-Variables

**Comment by kirelagin** [comment]
Reported upstream: http://lists.infradead.org/pipermail/pcsclite-muscle/2021-December/001215.html

**Comment by kirelagin** [comment]
Fixed in 1.9.5 (just released): https://github.com/LudovicRousseau/PCSC/commit/88ec131029ba9648a63ca072004e90414d10647e

**Comment by peterhoeg** [comment]
Keep in mind that this needs to point at `/etc/pcscd` if it is to be used for anything instead of into the nix store.

**Comment by VergeDX** [comment]
Can we make an option to allow specific `pkgs.pcsclite`?
I can help to make a PR if you agree : ) @kirelagin

**Comment by VergeDX** [comment]
The patch:
```````diff
diff --git a/nixos/modules/services/hardware/pcscd.nix b/nixos/modules/services/hardware/pcscd.nix
index b1a5c680a02..5f103053d80 100644
--- a/nixos/modules/services/hardware/pcscd.nix
+++ b/nixos/modules/services/hardware/pcscd.nix
@@ -10,6 +10,7 @@ let
     paths = map (p: "${p}/pcsc/drivers") config.services.pcscd.plugins;
   };
 
+  cfg = config.services.pcscd;
 in
 {
 
@@ -17,6 +18,7 @@ in
 
   options.services.pcscd = {
     enable = mkEnableOption "PCSC-Lite daemon";
+    package = mkEnableOption pkgs.pcsclite;
 
     plugins = mkOption {
       type = types.listOf types.package;
@@ -50,8 +52,8 @@ in
 
     environment.etc."reader.conf".source = cfgFile;
 
-    environment.systemPackages = [ pkgs.pcsclite ];
-    systemd.packages = [ (getBin pkgs.pcsclite) ];
+    environment.systemPackages = [ cfg.package ];
+    systemd.packages = [ (getBin cfg.package) ];
 
     systemd.sockets.pcscd.wantedBy = [ "sockets.target" ];
 
@@ -67,7 +69,7 @@ in
       # around it, we force the path to the cfgFile.
       #
       # https://github.com/NixOS/nixpkgs/issues/121088
-      serviceConfig.ExecStart = [ "" "${getBin pkgs.pcsclite}/bin/pcscd -f -x -c ${cfgFile}" ];
+      serviceConfig.ExecStart = [ "" "${getBin cfg.package}/bin/pcscd -f -x -c ${cfgFile}" ];
     };
   };
 }
```````

**Comment by kirelagin** [comment]
@VergeDX Send a PR I guess? Although, I am not sure how useful this option will be, given that, I assume, all other packages have to be linked to the same pcsc that provides the daemon. On the other hand, many other packages have this option, and I don’t know if there is some pattern or policy in NixOS.

**Comment by VergeDX** [comment]
The fixes PR is merged, but if a stable user (`nixos-21.11`, etc.) wants to solve these issues, he could be:
1. Use nix overlays, let `pcsclite` = the package from newer channel, this will cause rebuild of all package that depends on it.
2. Use `lib.mkForce` to override attribute of `systemd.service.pcscd`, which is tedious and error-prone.

So I suggest either we add an option to allow specific the package, or backport the changes to stable branch.

**Comment by VergeDX** [comment]
@kirelagin
